### PR TITLE
Add numba to speed dependencies

### PR DESCRIPTION
**Pull request by jlaehne**
Add numba to speed dependencies to fix failing tests and align minimum requirements with those of hyperspy.

### CODE REVIEW

**Comment by jlaehne** [comment]
pre-commit autoupdate

**Comment by jlaehne** [comment]
The lazy EELS fitting and background fitting is returning `nan` values in certain tests leading to a failure. However, this appears only from `python 3.10` and not for earlier python versions.

affected tests:
`exspy/tests/models/test_eelsmodel.py::TestFitBackground2D::test_lazy_only_current_false`
`exspy/tests/models/test_linear_model.py::TestLinearEELSFitting::test_lazy_convolved_and_std_error`
`exspy/tests/models/test_linear_model.py::TestLinearEELSFitting::test_lazy_nonconvolved`

@ericpre @CSSFrancis : Will this be fixed by https://github.com/hyperspy/hyperspy/pull/3476 ?

**Comment by ericpre** [comment]
> @ericpre @CSSFrancis : Will this be fixed by [hyperspy/hyperspy#3476](https://github.com/hyperspy/hyperspy/pull/3476) ?

Mostly, but no entirely: there is a something else going on with the fitting convolution code, I started to have a look at it, but it turns out to be not straightforward and take time. The convolution is implemented in exspy, not hyperspy and this is why this hasn't been done in https://github.com/hyperspy/hyperspy/pull/3476.

File: pyproject.toml [hunk 1]
```diff
@@ -73,7 +73,10 @@ file = "LICENSE"
   # unpin when sphinxcontrib-towncrier support more recent version to towncrier
   "towncrier<24",
 ]
-"speed" = ["numexpr"]
+speed = [
+    "numba>=0.53",
+    "numexpr>=2.8",
```
Comment (ericpre) [inline pyproject.toml:78]:
```suggestion
    "hyperspy[speed]",
```

**Review by ericpre — APPROVED** [review]
